lang/tests/src/test/scala/com/wavesplatform/lang/compiler/ExpressionCompilerV1Test.scala

```diff
@@ -51,6 +51,62 @@ class ExpressionCompilerV1Test extends PropSpec with PropertyChecks with Matcher
     }
   }
 
+  property("string limit") {
+    val maxString = "a" * Terms.DataEntryValueMax
+    val expr = Parser.parseExpr(s""" "$maxString" """).get.value
+    ExpressionCompiler(compilerContext, expr).map(_._1) shouldBe CONST_STRING(maxString)
+
+    val tooBigString = maxString + "a"
+    val expr2 = Parser.parseExpr(s""" "$tooBigString" """).get.value
+    ExpressionCompiler(compilerContext, expr2) should produce("String size=32768 exceeds 32767 bytes")
+
+  }
+
+  property("expression compilation fails if function name length is longer than 255 bytes") {
+    val tooLongName = "a" * (ContractLimits.MaxDeclarationNameInBytes + 1)
+    val funcExpr = {
+      val script =
+        s"""
+           |func $tooLongName() = 1
+           |true
+        """.stripMargin
+      Parser.parseExpr(script).get.value
+    }
+    val letExpr = {
+      val script =
+        s"""
+           |let $tooLongName = 1
+           |true
+        """.stripMargin
+      Parser.parseExpr(script).get.value
+    }
+    ExpressionCompiler(compilerContext, funcExpr) should produce("exceeds 255")
```

Comment:
Will it be visible in the text of this error how much the limit is exceeded?

Reply:
Fixed